web/client-api/src/main/java/io/deephaven/web/client/api/WorkerConnection.java

```diff
@@ -907,38 +961,84 @@ private TicketAndPromise<?> exportScopeTicket(JsVariableDefinition varDef) {
+            legacyResponse.setTypedExportIdsList(
+                    Arrays.stream(widget.getExportedObjects()).map(JsWidgetExportedObject::takeTicket)
+                            .toArray(TypedTicket[]::new));
+            c.apply(null, legacyResponse);
+            widget.close();
+            return null;
+        }, error -> {
+            c.apply(error, null);
+            return null;
+        });
     }
 
-    public Promise<JsFigure> getFigure(JsVariableDefinition varDef) {
-        if (!varDef.getType().equalsIgnoreCase("Figure")) {
-            throw new IllegalArgumentException("Can't load as a figure: " + varDef.getType());
-        }
+    private Promise<JsFigure> getFigure(Promise<JsWidget> widgetPromise) {
         return whenServerReady("get a figure")
-                .then(server -> new JsFigure(this,
-                        c -> {
-                            getWidget(varDef).then(widget -> {
-                                FetchObjectResponse legacyResponse = new FetchObjectResponse();
-                                legacyResponse.setData(widget.getDataAsU8());
-                                legacyResponse.setType(widget.getType());
-                                legacyResponse.setTypedExportIdsList(Arrays.stream(widget.getExportedObjects())
-                                        .map(JsWidgetExportedObject::typedTicket).toArray(TypedTicket[]::new));
-                                c.apply(null, legacyResponse);
-                                return null;
-                            }, error -> {
-                                c.apply(error, null);
-                                return null;
-                            });
-                        }).refetch());
+                .then(server -> new JsFigure(this, getWidgetFigureFetch(widgetPromise)).refetch());
+
+    }
+
+    public Promise<JsFigure> getFigure(JsVariableDefinition varDef) {
+        return getFigure(getWidget(varDef));
+    }
+
+    public Promise<JsFigure> getFigure(TypedTicket ticket) {
+        return getFigure(getWidget(ticket, true));
+    }
+
+    /**
+     * Gets a Pandas Table from the Pandas table widget provided. Closes the widget when done.
+     * 
+     * @param widget Pandas table widget
+     * @return Table representing the Pandas data frame
+     */
+    private Promise<JsTable> getPandasTable(JsWidget widget) {
+        TypedTicket typedTicket = widget.getExportedObjects()[0].takeTicket();
+
+        // We don't need to keep the original widget open, we can just close it right away
+        widget.close();
+
+        return getTable(typedTicket, "table for pandas").then(table -> {
+            // TODO(deephaven-core#3604) if using a new session don't attempt a reconnect
+            // never attempt a reconnect, since we might have a different widget schema entirely
+            table.addEventListener(JsTable.EVENT_DISCONNECT, ignore -> table.close());
+            return Promise.resolve(table);
+        });
```

Comment:
Is this necessary? Can't we just do something sorta like

```suggestion
        return widget.getExportedObjects()[0].fetch(true).then(table -> {
            // TODO(deephaven-core#3604) if using a new session don't attempt a reconnect
            // never attempt a reconnect, since we might have a different widget schema entirely
            table.addEventListener(JsTable.EVENT_DISCONNECT, ignore -> table.close());
            return Promise.resolve(table);
        });
```

Reply:
We still need to close the `JsWidget`. Maybe could have a method on JsWidget like `Promise<?> takeObjects()` which returns a `Promise.all` of all the exported objects fetch objects and closes the parent widget as well?

Reply:
Now we are releasing the ticket when the widget gets the close event from the server, so this is no longer necessary.